Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -274,3 +274,118 @@ describe('VirtualizedList', () => {
     );
   });
 });
+
+describe('VirtualizedList > OnEndReached', () => {
+  const ITEM_HEIGHT = 100;
+  const APPENDED_ITEM_COUNT = 10;
+
+  let listItems, onEndReached, instance;
+
+  beforeEach(() => {
+    listItems = appendNewItems([], 10);
+
+    onEndReached = jest.fn(function() {
+      appendNewItems(listItems, APPENDED_ITEM_COUNT);
+    });
+
+    instance = createComponentInstance();
+  });
+
+  it('should not be called after initial rendering', () => {
```

Comment:
Why wouldn't you want it to be called? The common case is to initially only fetch enough data to fill the viewport, then fetch more to preload the content below the fold so it's immediately available as soon as the user scrolls down to see it. The only case where `onEndReached` should not be called on initial scroll is if you overfetch a bunch of content initially, or you set the `onEndReachedThreshold` very small, neither of which is recommended for optimal performance.

This can be verified by this very test case - if you increase the initial number of items to 30, or drop the `onEndReachedThreshold` to 0.5 (note the default is 2), there will no longer be an initial call to `onEndReached` and this test will pass.

Reply:
For example, if 20 items are initially loaded
- contentLength = 2000
- visibleLength = 736

The height of all content view(contentLength) is close to three times the height of visible content view(visibleLength). In this case, since there is enough scrolling area, no one would want to load additional data for performance on initial rendering. In the general case, we would try to increase the number of initial data to avoid any performance degradation that may occur with scrolling.

Of course, as you said, we can fix this by adjusting `onEndReachedThreshold` to less than 1. But, finding the best threshold value in each case is difficult and cumbersome. The another issue is that reducing `onEndReachedThreshold` may have side effects at the point where additional data is loaded.